tests/standalone_tests/test_node_info_tasks_priority.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+
+from mipengine.controller.celery_app import CeleryAppFactory
+from mipengine.controller.celery_app import CeleryTaskTimeoutException
+from mipengine.controller.node_info_tasks_handler import NodeInfoTasksHandler
+from mipengine.node import config as node_config
+from mipengine.node_info_DTOs import NodeInfo
+from mipengine.node_tasks_DTOs import NodeTableDTO
+from mipengine.node_tasks_DTOs import UDFKeyArguments
+from mipengine.node_tasks_DTOs import UDFPosArguments
+from mipengine.udfgen import make_unique_func_name
+from tests.algorithms.orphan_udfs import one_second_udf
+from tests.standalone_tests.conftest import RABBITMQ_GLOBALNODE_ADDR
+from tests.standalone_tests.nodes_communication_helper import get_celery_task_signature
+from tests.standalone_tests.test_udfs import create_table_with_one_column_and_ten_rows
+
+request_id = "TestingSystemCallPriority"
+command_id = 0
+
+
+def queue_one_second_udf(
+    cel_app,
+    input_table_name,
+    logger,
+):
+    run_udf_task = get_celery_task_signature("run_udf")
+
+    kw_args_str = UDFKeyArguments(
+        args={"table": NodeTableDTO(value=input_table_name)}
+    ).json()
+
+    global command_id
+    command_id += 1
+    return cel_app.queue_task(
+        task_signature=run_udf_task,
+        logger=logger,
+        request_id=request_id,
+        command_id=str(command_id),
+        context_id=request_id,
+        func_name=make_unique_func_name(one_second_udf),
+        positional_args_json=UDFPosArguments(args=[]).json(),
+        keyword_args_json=kw_args_str,
+    )
+
+
+@pytest.mark.slow
+def test_node_info_tasks_have_higher_priority_over_other_tasks(
+    globalnode_node_service,
+    reset_celery_app_factory,
+    get_controller_testing_logger,
+):
+    cel_app_wrapper = CeleryAppFactory().get_celery_app(RABBITMQ_GLOBALNODE_ADDR)
+
+    input_table_name, _ = create_table_with_one_column_and_ten_rows(
+        cel_app_wrapper, request_id
+    )
+
+    # Queue an X amount of udfs to fill the rabbitmq.
+    # The queued udfs should be greater than the NODE workers that consume them.
+    number_of_udfs_to_schedule = node_config.celery.worker_concurrency + 10
+    udf_async_results = [
+        queue_one_second_udf(
+            cel_app_wrapper, input_table_name, get_controller_testing_logger
+        )
+        for _ in range(number_of_udfs_to_schedule)
+    ]
+
+    # The node info task should wait for one udf to complete, in order to start being executed,
+    # but shouldn't wait for more than one, since it has priority of the other tasks.
```

Comment:
With these comments node_info_task_timeout = 2 would be sufficient.
Is there a reason it is 3?